spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,579 @@
+            List<ZipEntry> consumerTypeZipFiles = ExportUtil.getFilesInDirectory(CONSUMER_TYPE_PATH, export);
+            Map<String, File> actual = ExportUtil.copyToTempFiles(export, consumerTypeZipFiles);
+            List<ConsumerTypeDTO> actualConsumerTypes = new ArrayList<>();
+            for (File jsonFile : actual.values()) {
+                assertNotNull(jsonFile);
+                ConsumerTypeDTO type = ConsumerTypeDTO.fromJson(Files.readString(jsonFile.toPath()));
+                if (type != null) {
+                    actualConsumerTypes.add(type);
+                }
+            }
+
+            assertTrue(CollectionUtils.isEqualCollection(expectedConsumerTypes, actualConsumerTypes));
+        }
+
+        @Test
+        @DisplayName("should export consumers")
+        void shouldExportConsumers() throws Exception {
+            ZipEntry consumerZipFile = ExportUtil.getFile("consumer.json", export);
+            assertNotNull(consumerZipFile);
+            Map<String, File> entryNameToTempFile = ExportUtil
+                .copyToTempFiles(export, List.of(consumerZipFile));
+            File consumerFile = entryNameToTempFile.get(consumerZipFile.getName());
+
+            JsonNode root = new ObjectMapper().readTree(Files.readString(consumerFile.toPath()));
+            String actual = root.get("uuid").asText();
+            assertEquals(consumer.getUuid(), actual);
+        }
+
+        @Test
+        @DisplayName("should exports CDN URL")
+        void shouldExportCdnUrl() throws Exception {
+            ZipEntry metaFile = ExportUtil.getFile("meta.json", export);
+            assertNotNull(metaFile);
+            Map<String, File> entryNameToTempFile = ExportUtil.copyToTempFiles(export, List.of(metaFile));
+            File metaTempFile = entryNameToTempFile.get(metaFile.getName());
+
+            JsonNode root = new ObjectMapper().readTree(Files.readString(metaTempFile.toPath()));
+            String actual = root.get("cdnLabel").asText();
+            assertEquals(cdn.getLabel(), actual);
+        }
+
+        @Test
+        @DisplayName("should not include consumer json in entitlements")
+        void shouldNotIncludeConsumerJsonInEntitlements() throws Exception {
+            List<ZipEntry> entitlements = ExportUtil.getFilesInDirectory(ENTITILEMENTS_PATH, export);
+            Map<String, File> entitlementFiles = ExportUtil.copyToTempFiles(export, entitlements);
+            for (File jsonFile : entitlementFiles.values()) {
+                JsonNode root = new ObjectMapper().readTree(Files.readString(jsonFile.toPath()));
+                assertNull(root.get("consumer"));
+            }
+        }
+
+        @Test
+        @DisplayName("should export candlepin 0.x backward compatible product definitions")
+        void shouldExportCandlepinBackwardCompatibleProductDefinitions() throws Exception {
+            List<ZipEntry> entitlements = ExportUtil.getFilesInDirectory(ENTITILEMENTS_PATH, export);
```

Comment:
This comment applies to this test and several others, but a lot of this code could be streamlined with use of streams directly from the zip file:

```
this.export.stream()
  .filter(entry -> entry.isFile() && entry.getName().beginsWith(ENTITLEMENTS_PATH))
  .forEach(entry -> {
    // JSON Node processing
  })
```

Reply:
Should we use the getFilesInDirectory method to also make sure that the zip entries are not in a sub directory?